otter/convergence/steps.py

```diff
@@ -222,4 +228,60 @@ def as_effect(self):
         Produce a :obj:`Effect` to remove some nodes from some RCv3 load
         balancers.
         """
-        return _rackconnect_bulk_request(self.lb_node_pairs, "DELETE", (204,))
+        eff = _rackconnect_bulk_request(self.lb_node_pairs, "DELETE",
+                                        success_pred=_CONSTANTLY_TRUE)
+        return eff.on(_rcv3_check_bulk_delete).on(_maybe_execute_step)
+
+
+_CONSTANTLY_TRUE = lambda _: True
+_UUID4_REGEX = ("[a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}"
+                "-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}")
+_RCV3_NODE_NOT_A_MEMBER_PATTERN = re.compile(
+    "Node (?P<node_id>{uuid}) is not a member of Load Balancer Pool "
+    "(?P<lb_id>{uuid})".format(uuid=_UUID4_REGEX),
+    re.IGNORECASE)
+
+
+def _rcv3_check_bulk_delete(result):
+    """
+    Checks if the RCv3 bulk deletion command was successful.
+
+    The request is considered successful if the response code
+    indicated unambiguous success, or the machine we're trying to
+    remove is already removed anyway.
+
+    If the machine was already removed, returns the next step to try
+    and remove the remaining pairs, unless a fatal error *also*
+    occurred. This is necessary because RCv3 bulk requests are
+    atomic-ish.
+    """
+    response, body = result
+
+    if response.code == 204:  # All done!
+        return
+
+    pairs_to_delete = []
+    for error in body["errors"]:
+        match = _RCV3_NODE_NOT_A_MEMBER_PATTERN.match(error)
+        if not match:  # Unrecoverable error, bail!
+            return
+
+        node_id, lb_id = match.groups()
+        pairs_to_delete.append((lb_id, node_id))
+
+    # REVIEW: cc @radix The best thing to return here would be an
+    # effect; that way it would get executed immediately. However, I
+    # want something that's better to introspect for testing, and
+    # that's definitely the step, not the effect.
```

Comment:
I don't understand why you think the Effect isn't introspectable for testing, we do testing with Effects everywhere. The intent should support equality, so you just need to check to see if it's what you expect.